stglib/iq.py

```diff
@@ -284,14 +372,362 @@ def cdf_to_nc(cdf_filename):
+                .replace("Profile_0_bindepth", "Profile_bindepth1")
+            )
+        elif "Profile_1" in var:
+            newvars[var] = (
+                var.replace("Profile_1_Amp", "Profile_AGC2_1222")
+                .replace("Profile_1_Vel", "Profile_vel2_1278")
+                .replace("Profile_1_BlankingDistance", "Profile_blanking_distance2")
+                .replace("Profile_1_CellSize", "Profile_bin_size2")
+                .replace("Profile_1_bindist", "Profile_bindist2")
+                .replace("Profile_1_z", "Profile_z2")
+                .replace("Profile_1_bindepth", "Profile_bindepth2")
+            )
+        elif "Profile_2" in var:
+            newvars[var] = (
+                var.replace("Profile_2_Amp", "Profile_AGC3_1223")
+                .replace("Profile_2_Vel", "Profile_vel3_1279")
+                .replace("Profile_2_BlankingDistance", "Profile_blanking_distance3")
+                .replace("Profile_2_CellSize", "Profile_bin_size3")
+                .replace("Profile_2_bindist", "Profile_bindist3")
+                .replace("Profile_2_z", "Profile_z3")
+                .replace("Profile_2_bindepth", "Profile_bindepth3")
+            )
+        elif "Profile_3" in var:
+            newvars[var] = (
+                var.replace("Profile_3_Amp", "Profile_AGC4_1224")
+                .replace("Profile_3_Vel", "Profile_vel4_1280")
+                .replace("Profile_3_BlankingDistance", "Profile_blanking_distance4")
+                .replace("Profile_3_CellSize", "Profile_bin_size4")
+                .replace("Profile_3_bindist", "Profile_bindist4")
+                .replace("Profile_3_z", "Profile_z4")
+                .replace("Profile_3_bindepth", "Profile_bindepth4")
+            )
+
+    varnames = {
+        "Batt": "Bat_106",
+        "Temp": "T_28",
+        "Pitch": "Ptch_1216",
+        "Roll": "Roll_1217",
+        "Depth": "D_3",
+        "Pressure": "P_1",
+        "AdjustedPressure": "P_1ac",
+        "SoundSpeed": "SV_80",
+    }
+    # check to make sure they exist before trying to rename
+    for k in varnames:
+        if k in ds:
+            newvars[k] = varnames[k]
+
+    return ds.rename(newvars)
+
+
 def ds_add_attrs(ds):
-    ds.attrs["serial_number"] = ds.attrs["SerialNumber"]
-    ds.attrs["INST_TYPE"] = "SonTek-IQ Plus"
+    attrsnams = ["InstrumentSubType", "InstrumentFriendlyName"]
+
+    for k in attrsnams:
+        del ds.attrs[k]
+
+    ds.attrs["serial_number"] = ds.attrs.pop("SerialNumber")
+    ds.attrs["instrument_type"] = (
+        ds.attrs.pop("InstrumentFamily") + "-" + ds.attrs.pop("InstrumentType")
+    )
+
+    if "positive_direction" not in ds.attrs:
+        ds.attrs["positive_direction"] = "Not specified"
+        print(
+            "Define positive_direction attribute in yaml file (direction of x arrow on IQ at field site)."
+        )
+
+    if "flood_direction" not in ds.attrs:
+        ds.attrs["flood_direction"] = "Not specified"
```

Comment:
See questions above.

Reply:
Because there are different conventions for positive vel = flood vs positive vel = ebb, I wanted to specify the direction of flood velocities. For this timeseries release I am working on, it was requested that I set up the IQ so that positive velocities were ebb. After talking with Steve, we decided it was a good idea to specify the direction of positive x velocities and also the flood direction on site. 

For now, I will leave this as an option so just adding a warning.